src/utils/NovorAdapter.cpp

```diff
@@ -0,0 +1,389 @@
+    // mass error tolerance
+    registerDoubleOption_("fragmentIon_error_tolerance", "<double>", 0.5, "Fragmentation error tolerance  (Da)", false);
+    registerDoubleOption_("precursor_error_tolerance", "<double>" , 15.0, "Precursor error tolerance  (ppm or Da)", false);
+    registerStringOption_("precursor_error_units", "<choice>", "Da", "Unit of precursor mass tolerance", false);
+    setValidStrings_("precursor_error_units", ListUtils::create<String>("Da,ppm"));
+    // post-translational-modification
+    registerStringList_("variable_modifications", "<mods>", vector<String>(), "Variable modifications", false);
+    setValidStrings_("variable_modifications", ListUtils::create<String>("Acetyl (K),Acetyl (N-term),Amidated (C-term),Ammonia-loss (N-term C),Biotin (K),Biotin (N-term),Carbamidomethyl (C),Carbamyl (K),Carbamyl (N-term),Carboxymethyl (C),Deamidated (NQ),Dehydrated (N-term C),Dioxidation (M),Methyl (C-term),Methyl (DE),Oxidation (M),Oxidation (HW),Phospho (ST),Phospho (Y),Pyro-carbamidomethyl (N-term C),Pyro-Glu (E),Pyro-Glu (Q),Sodium (C-term),Sodium (DE),Sulfo (STY),Trimethyl (RK)"));
+    registerStringList_("fixed_modifications", "<mods>", vector<String>(), "Fixed modifications", false);
+    setValidStrings_("fixed_modifications", ListUtils::create<String>("Acetyl (K),Acetyl (N-term),Amidated (C-term),Ammonia-loss (N-term C),Biotin (K),Biotin (N-term),Carbamidomethyl (C),Carbamyl (K),Carbamyl (N-term),Carboxymethyl (C),Deamidated (NQ),Dehydrated (N-term C),Dioxidation (M),Methyl (C-term),Methyl (DE),Oxidation (M),Oxidation (HW),Phospho (ST),Phospho (Y),Pyro-carbamidomethyl (N-term C),Pyro-Glu (E),Pyro-Glu (Q),Sodium (C-term),Sodium (DE),Sulfo (STY),Trimethyl (RK)"));
+   // forbidden residues
+   registerStringList_("forbiddenResidues", "<mods>", vector<String>(), "Forbidden Resiudes", false);
+   setValidStrings_("forbiddenResidues", ListUtils::create<String>("I,U"));
+ 
+   // parameter novorFile will not be wrapped here
+   
+   registerInputFile_("java_executable", "<file>", "java", "The Java executable. Usually Java is on the system PATH. If Java is not found, use this parameter to specify the full path to Java", false, false, ListUtils::create<String>("skipexists"));
+   registerIntOption_("java_memory", "<num>", 3500, "Maximum Java heap size (in MB)", false);
+
+  }
+
+  // remove temporary folder 
+  void removeTempDir_(const String& tmp_dir)
+  {
+    if (tmp_dir.empty()) {return;} // no temporary directory created
+
+    if (debug_level_ >= 2)
+    {
+      writeDebug_("Keeping temporary files in directory '" + tmp_dir + "'. Set debug level to 1 or lower to remove them.", 2);
+    }
+    else
+    {
+      if (debug_level_ == 1) 
+      {
+        writeDebug_("Deleting temporary directory '" + tmp_dir + "'. Set debug level to 2 or higher to keep it.", 1);
+      }
+      File::removeDirRecursively(tmp_dir);
+    }
+  }
+
+  void createParamFile_(ostream& os)
```

Comment:
I think it may make sense to allow advanced users to just pass a param file...we also do this for other adapters

Reply:
One problem with that is that we need to parse the param file manually to properly annotate the search engine settings. 
If the param file format changes, this easily leads to difficult / undetected errors. Note that creating the param file for a wrong search engine version is usually more easily detected if it is done from tools parameters.

Reply:
I see the problem. I agree parsing the param file is not what we want to do. Still it is annoying if there is some parameter you would like to use and simply cant because its not supported by OpenMS. Maybe an alternative would be to 

* pass a param file and in that case ignore any options passed to OpenMS?
* pass a file whose content simply gets appended to the param file written by OpenMS

in either case I agree we dont want to parse the file. But I am also fine to leave it as is